Bug 1199078 - Add Help menu and useful menu entries

tojonmz opened the pull request:
This fixes the UI side of Bugzilla bug [1199078](https://bugzilla.mozilla.org/show_bug.cgi?id=1199078).

This adds a drop down menu for the Help icon for access to feature development news, and provides a variety of other useful entries for new/experienced users. The actual URL of the feed for Treeherder development news is still tbd. Presently it just points to the ATeam blog.

Current (no menu):
![current](https://cloud.githubusercontent.com/assets/3660661/9471852/44439d4c-4b20-11e5-908a-ee99f7eb56d0.jpg)

Proposed (dropdown added):
![proposed](https://cloud.githubusercontent.com/assets/3660661/9471867/5750da26-4b20-11e5-9b7c-81dfc8705481.jpg)

Proposed (dropdown open):
![proposed2](https://cloud.githubusercontent.com/assets/3660661/9471891/6dfc0ea8-4b20-11e5-997c-7d1b74e40f11.jpg)

I'm actually inclined to just call the 2nd entry "Developer Guide". I also left WhatsDeployed in the UserGuide footer, since users have experience with it being there, and I don't think it does harm to have it in two places.

The notable mappings are:
* User Guide (formerly help.html)
* Development Documentation (readthedocs)
* Development News (our feed, tbd)

Everything seems fine in local testing.

Tested on OSX 10.10.3:
Nightly **43.0a1 (2015-08-24)**
Chrome Latest Release **44.0.2403.157 (64-bit)**

Adding @edmorley for review.

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/mozilla/treeherder/906)
<!-- Reviewable:end -->

tojonmz commented:
Review status: 0 of 6 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

<sup>**[ui/partials/main/thGlobalTopNavPanel.html, line 75 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/906#-Jx_h4W2KVhe_5La1cVU)** ([raw file](https://github.com/mozilla/treeherder/blob/0efb43a9964d4f60235261d9b5a30d16812ac776/ui/partials/main/thGlobalTopNavPanel.html#L75)):</sup>
This `repo-menu-` here is intentional. We could generalize this name later, since we're using it to tighten the spacing of most of the r.h. navbar buttons.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/906)
<!-- Sent from Reviewable.io -->

tojonmz commented:
Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

<sup>**[ui/partials/main/thHelpMenu.html, line 27 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/906#-JxbQOcuWJbm4cjmrmAv)** ([raw file](https://github.com/mozilla/treeherder/blob/0efb43a9964d4f60235261d9b5a30d16812ac776/ui/partials/main/thHelpMenu.html#L27)):</sup>
I'll update this to the new What's Deployed url per bug 1198452, post review.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/906)
<!-- Sent from Reviewable.io -->

edmorley commented:
This is awesome :-)

Few thoughts:
* Let's omit the blog part for now, until we have a Treeherder category on the a-team planet and a few posts to see
* Let's add a "File a bug" or "Report a bug" link
* I'm not sure whether I prefer "Help", "User Guide" or "Help & Keyboard shortcuts" etc. I don't think the page it links to is currently a user guide (it's not that detailed), nor do we want it to be too detailed in the future I think? I wonder if a less heavyweight name might get people to click on it more?
* For the link to github, I wonder if "Source" might be better? Seeing as we don't use Github for issues + the canonical place for project info is the wiki + the canonical place for dev docs is read the docs - then Github is really just about the source. The Github icon also implies the source is on Github, for people who might be attracted by the thought of git and/or Github vs hg.m.o.
* Could you rebase on master 

Thanks!

camd commented:
Yeah, this is really great!  I wonder if "cheat sheet" or "quick reference" would be a better name for our current Help page.

tojonmz commented:
> Yeah, this is really great! I wonder if "cheat sheet" or "quick reference" would be a better name for our current Help page.

Mostly I chose 'User Guide" to differentiate that document targeting an end-user of the UI, vs. from a developer (who would select "Development Documentation"). I could also see User Guide growing in content size a bit, as lately I've been thinking of adding illustrations and explanations of basic parts of the UI into that doc. I know it's been static for some time.

tojonmz commented:
> For the link to github, I wonder if "Source" might be better?

I could switch it to "Github Source" if you like? I figured having "Github" in there somewhere always helps keep things clear for newbies, and for screen readers so you know where you'll land. I'll add a "File a Bug" b.m.o link, that will also make clear Github is not the entire project.

tojonmz commented:
nb. we're switching this PR to bug [1199078](https://bugzilla.mozilla.org/show_bug.cgi?id=1199078) to land all changes except the planet.m.o feed addition, so we don't block on it. I will change the subject and commit accordingly.

edmorley commented:
> I could switch it to "Github Source" if you like? I figured having "Github" in there somewhere always helps keep things clear for newbies, and for screen readers so you know where you'll land.

"Github Source" implies "the source of Github", if we want to include the name Github, I think "Source on Github" might be better. That said, given it has the github icon I wonder if "Source" is fine? For screen readers we can always use the `title` attribute to be more descriptive.

tojonmz commented:
Sure I'll go with just "Source" then and maybe try adding `title=` for everything. To me "Github Source" seemed to be fine in the context in which it is being presented, but I understand the interpretation.

I'm a fan of having the leading word in any menu being the biggest cue on what/where they will go, rather than having to read to the end of the menu string to comprehend it. My test is always to just glance at the first word in any multi-word menu line and see if it has the desired meaning. "User", "Development", "Development", and so forth. It's not always a perfect world though.

I might go with "Report a Problem" or "Report a Bug" in that vein, rather than 'File', also.

tojonmz commented:
Ok I've made the changes, and also rebased (but haven't squashed yet). I tried putting `title=` but the menu interaction on hover seemed a bit of information overkill. I think the menus themselves should be fine. I also added an IRC menu prior to the Report a Bug menu for users experiencing a problem (and who had a suitable IRC client) to get help there first. If we don't want that menu I can remove it.

![iteration1](https://cloud.githubusercontent.com/assets/3660661/9524695/4a140cb8-4cae-11e5-85ec-b72ded84a01e.jpg)

tojonmz commented:
I just needed another rebase over other Help changes I just landed, but the above commit `- it1` is the delta between the last review :)

edmorley commented:
Looking good :-)

Last round of changes:
* Let's drop the "Talk to us" link. I'd like to promote people filing bugs for issues rather than interrupting us (and causing context switching) for things that are feature requests etc. I also think this menu will look better with half a dozen entries rather than nearing 10 (once any blog link is added etc)
* I think "Report a Bug" should move above "Source" - since "What's Deployed" and source are more related, and I think reporting a bug is also a more common action (so should be sooner in the list)

Other things that needn't block landing / I'll leave up to you to decide:
* I know we settled on "What's Deployed?" for the footer link before, but I wonder if we can pick a more informative string for people who haven't heard of that tool (eg "Deployed revision", or ..?).
* I can't decide whether I prefer "Title Case" or "Sentence case".
* The text for "What's Deployed" is slightly mis-aligned since the icon is a different width

tojonmz commented:
Thanks Ed, I'm going to fix everything you've outlined in both categories except perhaps I think What's Deployed? as a menu item sounds reasonable to me, since that's the name of the service (similar to our Infra menus).

I'm also inclined to stay with "Title Case" for these, although I don't really have a reason other than they all launch to external pages and are quasi 'proper names'. Versus menus like the pinboard save menu, or resultset action bar menu, which are local treeherder menu functions. Kind of tenuous logic, but that was my thinking. We can always adjust though later if we want.

tojonmz commented:
Ok, here's the changes in `- it2` for posterity:

![it3](https://cloud.githubusercontent.com/assets/3660661/9555210/a8cef7e0-4d99-11e5-8c55-3fc7e00ccead.jpg)

edmorley commented:
Looks great :-)

tojonmz commented:
Cool, will squash and merge.

tojonmz commented:
**nb:** I stomped on the updated bug number during my rebase, I've updated the bugs accordingly but unfortunately the commit has the original bug number [1197988](https://bugzilla.mozilla.org/show_bug.cgi?id=1197988) instead of bug [1199078](https://bugzilla.mozilla.org/show_bug.cgi?id=1199078). For some reason I figured git rebase could read my mind :) Lesson learned.
